mutters-bert-intent/src/main/java/com/rabidgremlin/mutters/bert/doccat/DoccatModel.java

```diff
@@ -0,0 +1,141 @@
+package com.rabidgremlin.mutters.bert.doccat;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.tensorflow.SavedModelBundle;
+import org.tensorflow.Session;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Streams;
+
+/**
+ * Represents a loaded BERT document categorizer (doccat) model.
+ *
+ * @author wilmol
+ */
+public class DoccatModel
+    implements AutoCloseable
+{
+  private final Logger log = LoggerFactory.getLogger(DoccatModel.class);
+
+  private final Session modelSession;
+
+  private final int maxSeqLength;
+
+  private final ImmutableMap<Integer, String> modelLabels;
+
+  private final ImmutableMap<String, Long> modelVocab;
+
+  /**
+   * Constructor; loads the BERT model and associated files.
+   *
+   * @param modelPath path to the saved_model.pb file and variables folder
+   * @param maxSeqLength {@code max_seq_length} value used to train the model
+   * @param labelsPath path to the labels.txt file used to train the model
+   * @param vocabPath path to the vocab.txt file used to train the model
+   * @throws IllegalArgumentException if loading the model fails
+   */
+  public DoccatModel(String modelPath, int maxSeqLength, String labelsPath, String vocabPath)
+  {
+    log.debug("Loading model and creating session");
+    try
+    {
+      // TODO(wilmol) use Resources#getResource here?
+      SavedModelBundle model = SavedModelBundle.load(modelPath, "serve");
```

Comment:
Looks like they only support loading from file/disk. Makes sense since unlikely to bundle giant model into .jar file. Pity it cannot load from URL....